var/spack/repos/builtin/packages/fenics-dolfinx/package.py

```diff
@@ -40,7 +41,9 @@ class FenicsDolfinx(CMakePackage):
 
     variant("slepc", default=False, description="slepc support")
     variant("adios2", default=False, description="adios2 support")
+    variant("petsc", default=True, description="PETSc support")
```

Comment:
```suggestion
    variant("petsc", default=False, description="PETSc support")
```

Although this will be a modification to previously installed versions of the package, I think we should have `petsc` default to false instead of true. 

If packages depend on `fenics-dolfinx +petsc` they should be explicit to call out that dependency, which will allow us to reduce the complexity of environments for users who don't require petsc support.